### PR TITLE
[Tests] New iteration of removing `$this` occurrences in future static data providers

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes-ish
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Easing https://github.com/symfony/symfony/pull/48668#issuecomment-1379892415
| License       | MIT
| Doc PR        | _NA_

Follow-up of #48980

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Asset/Tests/VersionStrategy/JsonManifestVersionStrategyTest.php [hunk 1]
```diff
@@ -84,17 +84,17 @@ public function testRemoteManifestFileWithoutHttpClient()
...
+        yield from static::provideStrategies('manifest-invalid.json');
     }
 
     public function provideMissingStrategies()
     {
-        yield from $this->provideStrategies('non-existent-file.json');
+        yield from static::provideStrategies('non-existent-file.json');
     }
 
     public function provideStrategies(string $manifestPath)
```
Comment (OskarStark) [inline src/Symfony/Component/Asset/Tests/VersionStrategy/JsonManifestVersionStrategyTest.php]:
```suggestion
    public static function provideStrategies(string $manifestPath)
```
?

Comment (OskarStark) [reply src/Symfony/Component/Asset/Tests/VersionStrategy/JsonManifestVersionStrategyTest.php]: The other PR will only change this to `static` automatically, if it is directly used as a dataProvider somewhere, is that the case?

Comment (alexandre-daubois) [reply src/Symfony/Component/Asset/Tests/VersionStrategy/JsonManifestVersionStrategyTest.php]: I would say yes but isn't it a BC break to do so on a public method? 😕

Comment (nicolas-grekas) [reply src/Symfony/Component/Asset/Tests/VersionStrategy/JsonManifestVersionStrategyTest.php]: tests are not covered by the BC policy

Comment (OskarStark) [reply src/Symfony/Component/Asset/Tests/VersionStrategy/JsonManifestVersionStrategyTest.php]: `/Test/` directory is covered by BC, while `/Tests/` not 😃

Comment (alexandre-daubois) [reply src/Symfony/Component/Asset/Tests/VersionStrategy/JsonManifestVersionStrategyTest.php]: Oh I wasn't aware of this detail, thank you! Updated the PR 🙂

**Comment by nicolas-grekas** [comment]
Can you merge this one and #49054 please?

**Comment by alexandre-daubois** [comment]
That's merged 👍

**Comment by OskarStark** [comment]
I think this can be merged @nicolas-grekas

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @alexandre-daubois.

**Pull request merged**